### PR TITLE
Save xcode results as artifact in Buildkite

### PR DESCRIPTION
**Pull request by tiagomar**
Saving Xcode results with logs and screenshots in Buildkite is the only way to avoid guessing what happened when a UI Test fails. 

To test:
Check if UI Tests step has the tests results in Artifacts tab.

## Regression Notes
1. Potential unintended areas of impact
Buildkite.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Checked the CI.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17884-2f22e74</code>. IPA is available <a href='https://appcenter-filemanagement-distrib3ede6f06e.azureedge.net/b26586dc-4fda-40ab-8ed3-7a49a0f9a47a/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=9C18XHb61pPIIxSRq7DX1RjTwhIhrld8GhoIoBk%2Be%2FQ%3D&se=2022-02-05T14%3A10%3A47Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17884-2f22e74</code>. IPA is available <a href='https://appcenter-filemanagement-distrib2ede6f06e.azureedge.net/db8e091c-c9fa-4d20-82d9-95e3114ef713/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=q5WhA7CWb7pGmXD3agDyzS%2B%2BsIvNgRxC1XNrHe97DH0%3D&se=2022-02-05T14%3A12%3A14Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

File: .buildkite/commands/run-ui-tests.sh
Comment (AliSoftware) [inline .buildkite/commands/run-ui-tests.sh]: Looking at the logs I now see that the `Error: The command exited with status 1` message that gets printed in the log due to that might misleadingly make us think that this is the "Zipping test results" steps that has failed, as opposed to the tests, so probably worth adding another header before this to separate that `exit` and clarify in logs:

```suggestion
echo "--- 🚦 Report Tests Exit Status
exit $TESTS_EXIT_STATUS

```

**Review by AliSoftware — APPROVED** [review]
Changes look good and seems to be working as expected 👍 

The CI is currently failing but I think that might be due to a flaky timeOut error while running those UI tests — at least that's what's [the HTML test report (from the build artifacts) suggests](https://buildkite.com/organizations/automattic/pipelines/wordpress-ios/builds/4923/jobs/c4432fc4-37dd-431f-9dfe-664c70791412/artifacts/3be6d8ed-edff-4ac8-bb23-ef6ddc034696) — and not related to the changes in this PR per se (they might even pass if you retry the step)… so approving the changes 👍

**Review by pachlava — APPROVED** [review]
Awesome work @tiagomar ! Could download the log, open it in Xcode and see automatic screenshots. The UI tests failures are exactly the reason for this PR, so they should not be treated as blockers. Thank you! 👍

**Comment by tiagomar** [comment]
Thanks for reviewing it! The UI Tests failure could have been on purpose. =P At least it served to test the PR.
I'll rerun them before merging just in case.

**Pull request merged**
